near-contract-standards/src/non_fungible_token/payout/payout_impl.rs

```diff
@@ -0,0 +1,75 @@
+use super::NonFungibleTokenPayout;
+use crate::non_fungible_token::core::NonFungibleTokenCore;
+use crate::non_fungible_token::payout::*;
+use crate::non_fungible_token::NonFungibleToken;
+use near_sdk::{assert_one_yocto, env};
+use near_sdk::{require, AccountId, Balance, IntoStorageKey};
+use std::collections::HashMap;
+impl Royalties {
+    pub fn new<S>(key_prefix: S) -> Self
+    where
+        S: IntoStorageKey,
+    {
+        let temp_accounts: HashMap<AccountId, BasisPoint> = HashMap::new();
+        let this =
+            Self { key_prefix: key_prefix.into_storage_key(), accounts: temp_accounts, percent: 0 };
+        this.validate();
+        this
+    }
+    pub(crate) fn validate(&self) {
+        require!(self.percent <= 100, "royalty percent must be between 0 - 100");
+        require!(
+            self.accounts.len() <= 10,
+            "can only have a maximum of 10 accounts spliting royalties"
+        );
+        let mut total: BasisPoint = 0;
+        self.accounts.iter().for_each(|(_, percent)| {
+            require!(*percent <= 100, "each royalty should be less than 100");
+            total += percent;
+        });
+        require!(total <= 100, "total percent of each royalty split  must be less than 100")
+    }
+    pub fn create_payout(&self, balance: Balance, owner_id: &AccountId) -> Payout {
```

Comment:
It'd be good to add some unit tests for the payout calculation especially around edge cases (0 inputs, inputs that can produce results that overflow, returned payout sum is not higher than the balance)

Reply:
agreed.

Reply:
I don't think there is a way to overflow as we make sure that the total percent is at most 100.

Reply:
I'm not sure I follow. :) The `apply_percent` function can overflow I think.

Reply:
Yeah, you are right. Because it first gets multiplied by 100. I have introduced an overflow test already to check this.

Reply:
I would argue that we could change the behaviour though. Do something like
```
    match int.checked_mul(percent as u128) {
       Some(int) => int / 100,
       None => int / 100 * percent
     }
```
Then it will still be correct, but never overflowing.